Data/DataExport.lua

```diff
@@ -75,7 +75,15 @@ function CraftSim.DATAEXPORT:handlePlayerProfessionStatsV1(recipeData, operation
 			recipeData.stats.inspiration.percent = CraftSim.UTIL:GetInspirationPercentByStat(recipeData.stats.inspiration.value) * 100
 
 			-- matches a row of numbers coming after the % character and any characters in between plus a space, should hopefully match in every localization...
-			local _, _, bonusSkill = string.find(statInfo.ratingDescription, "%%.* (%d+)") 
+			if GetLocale() == CraftSim.CONST.LOCALES.TW then

```

Comment:
I think you confused something here,
This function is for the UI extraction (Like the details panel in the crafting frame) of the player stats

The extraction of the inspiration bonusskill from the description should actually be localization independent, cause it it just tries to match for a number with a following % character

If you were searching for the code that extracts the bonusskill percent bonus, like in the blacksmith epic hammer, this happens in the item stat extraction in:
`function CraftSim.DATAEXPORT:GetProfessionGearStatsByLink`
and this tries to match a tooltip line by the provided match string of the loaded localization

Reply:
I originally got this error:
...erface/AddOns/CraftSim/Pricing/ProfitCalculation.lua:17: attempt to perform arithmetic on field 'bonusskill' (a nil value)

then I found that it's due to this line 78 of not getting the bonusskills when inspiration proc.
Here are the debug variables I printed:
statInfo.ratingDescription: 你有27.8%的機率獲得靈感，在製作這個配方時得到130點額外技能。
bonusSkill: nil
So I added a few lines to get the 130 bonusSkill

As you mentioned it was because of regex not working on Chinese characters. So I tried to removed all the involved Chinese characters and take the string after the % to get the bonusSkill.

Reply:
![image](https://user-images.githubusercontent.com/21115928/211314582-37f73c51-880f-4b3c-84d6-5fdc04e7125c.png)
I tried this to try to get a locale independent logic by gsub-ing all the Chinese characters out of the string but it didn't work...

local _, _, bonusSkill = tonumber(statInfo.ratingDescription:gsub("[^\w\%\.]", ""):gsub(".*%%", ""))

Reply:
hmm so the regex does not recognize the chinese characters?

Reply:
Yes. I have found a new logic that can work on all locales. Please check the latest commit~